apps/lib/s_cb.c

```diff
@@ -1420,12 +1431,16 @@ static int security_callback_debug(const SSL *s, const SSL_CTX *ctx,
         {
             if (cert_md) {
                 int sig_nid = X509_get_signature_nid(other);
+
                 BIO_puts(sdb->out, OBJ_nid2sn(sig_nid));
             } else {
                 EVP_PKEY *pkey = X509_get0_pubkey(other);
                 const char *algname = "";
-                EVP_PKEY_asn1_get0_info(NULL, NULL, NULL, NULL,
-                                        &algname, EVP_PKEY_get0_asn1(pkey));
+
+                if (pkey != NULL) {
```

Comment:
I suppose this is really an impossible error condition. However please note the pkey value is used in the following call as well. So perhaps it would be just better to do `if (pkey == NULL) break;` ?

Reply:
I noticed the following function `EVP_PKEY_get_bits`, but inside it checks the value of pkey, if it is `NULL` then returns 0.

Reply:
The BIO_printf will output useless information. Perhaps outputting something like `No public key` would be more useful?